### ISSUE TITLE
Aggressive Flash no longer works due to YouTube change today

### ISSUE THREAD

**Issue by dinsdalepiranha** [YePpHa/YouTubeCenter#2436]
At least in some browsers (SeaMonkey on Linux here) YT has started blocking Flash as of earlier today:

https://productforums.google.com/d/msg/youtube/CUaTWvKhAuE/Z9ONjZ-xAgAJ

Maybe just a matter of some JavaScript tweaks to get it back, though?

**Comment by thinkpad4** [comment]
I hope @YePpHa adds a option to disable this html5 change

**Comment by Leifman27** [comment]
yes :( i can confirm.. cannot watch youtube in flash on firefox. aggressive flash = black video box. and normal flash ends up in just being html5 :'( sad sad day.
as i prefer to watch youtube with flash on firefox, and on chrome with html5 for 60fps videos.. so now i don't have that option anymore :'(

**Comment by Xan-Kun** [comment]
I can confirm, testes in FF, Opera and Chrome. It seems flash is gone for good.

**Comment by dinsdalepiranha** [comment]
I did notice that it still works in _embedded_ videos, however...

**Comment by Xan-Kun** [comment]
@dinsdalepiranha can you elaborate? I just tried with the standard (YT generated) html code and still end up with a nasty CPU killing <video> tag. :-)

**Comment by dinsdalepiranha** [comment]
@Xan-Kun With this code:
`<div style="margin-top:3px;">
		<iframe width="620" height="350" src="//www.youtube.com/embed/5r0Sh1--Hc4?rel=0&wmode=transparent" frameborder="0" allowfullscreen></iframe>
	</div>`

Embedded: http://i.imgur.com/Tb2DQtg.png
Same video directly on YouTube: http://i.imgur.com/7zXMZex.png

**Comment by digideth** [comment]
I doubt if there is anything that can be done.
There is a userscript on that google forum link above that supposedly embeds the video on the youtube page you are on so the flashplayer loads but I couldn't get it to work.

**Comment by Xan-Kun** [comment]
@dinsdalepiranha What Broser is that and which plugin/addon? I tried your code and still got to html5.
Did you load it from an actual server or was it a file:// url?

**Comment by dinsdalepiranha** [comment]
@digideth @Xan-Kun @Leifman27 @thinkpad4 The script in [this post of the above thread](https://productforums.google.com/d/msg/youtube/CUaTWvKhAuE/q6P1m1kpAwAJ) _does_ work, at least for me.  In YouTube Center settings make sure **External Players** > **Player Type** is set to Aggressive Flash then add this userscript to Greasemonkey or whatever:
```javascript
// ==UserScript==
// @name          YouTube Force Flash via Embed
// @namespace     YT_forceflashembed
// @description   Force Flash player by embedding YouTube in YouTube. From: https://productforums.google.com/d/msg/youtube/CUaTWvKhAuE/q6P1m1kpAwAJ
// @version 1
// @include       *.youtube.com/*
// @grant         GM_addStyle
// ==/UserScript==
// Notes:
//   * is a wildcard character
//   .tld is magic that matches all top-level domains (e.g. .com, .co.uk, .us, etc.)
/* ____80_character_separator________________________________________________ */
window.setTimeout(function() {
  var embedFrame = document.createElement("iframe");
  embedFrame.src = location.href.replace("watch?v=", "embed/");
  embedFrame.style = "width: 100%; height: 100%;";
  var player = document.getElementById("player-api");
  player.innerHTML = "";
  player.appendChild(embedFrame);
},
3000);
```
Note that all of your other "External Players" YT Center preferences will now be in effect on YouTube itself (autoplay, default resolution, etc.) so you can't have a separate set of prefs for embedded YT videos anymore.

**Comment by Chazzen01** [comment]
@dinsdalepiranha Just wanted to say thanks for that script. It works perfectly and has saved me from the many HTML5-induced FF crashes I have been suffering through.

**Comment by Leifman27** [comment]
@dinsdalepiranha oh wow... it actually worked 0__0
THANK YOU.

**Comment by Xan-Kun** [comment]
@dinsdalepiranha Thanks a lot, I got it to work! The trick for me was, to remove the ".user" part from the file name.

**Comment by Maurogch** [comment]
Thanks, I was waiting for an update, but it seems that we'll have to wait a bit more, thanks for sharing a workaround.

**Comment by Leifman27** [comment]
thank you @Maurogch that totally fixed it, an issue i noticed but had no idea how to fix :)

**Comment by MasterBLB** [comment]
Hey mates.
How to use that script for Opera 12.18 browser?

**Comment by Xan-Kun** [comment]
@MasterBLB I just put it in the "User Javascript Folder" and make sure, it ends only with ".js". As a ".user.js" extension will not have sufficient rights.

**Comment by thinkpad4** [comment]
The workaround seems to have stopped working, anyone else have a workaround?

**Comment by Xan-Kun** [comment]
Same here.

**Comment by Leifman27** [comment]
:'( yep can confirm doesn't work anymore
